src/en/config-azure.md

```diff
@@ -1,123 +1,223 @@
+```
+
+Confirm it's installed correctly by viewing its online help. Then put it in
+*Azure Resource Manager* mode and log in:
+
+```bash
+azure help
+azure config mode arm
+azure login
 ```
 
-This will generate a file, `environments.yaml`, which will live in your
-`~/.juju/` directory (and will create the directory if it doesn't already
-exist).
+You will be prompted to visit a website to enter the provided code. It will
+therefore be easier to perform this on a (graphical) Ubuntu desktop.
+
+
+# Configuring for Microsoft Azure
 
-**Note:** The above command will not overwrite your existing environments.yaml
-file, or output to stdout. In order to see the boilerplate environments.yaml on
-stdout you need to append the `--show` option. This is helpful if you have an
-existing environments.yaml and just need to add a section. For example:
+If this is a new Juju install then you do not yet have a
+`~/.juju/environments.yaml` file. Create one with
+
+```bash
+juju generate-config
+```
+
+If it does exist (but it was created with an older version of Juju), first move
+it out of the way (back it up) and *then* generate a new one. Alternatively,
+you can output a generic file to screen (STDOUT) and paste the Azure parts into
+your existing file:
 
 ```bash
 juju generate-config --show
 ```
 
-You can then copy and paste the needed section.
+The file will contain a section for the Azure provider.
+
+
+## Configure and bootstrap
+
+Values will need to be found for the following parameters:
+
+ - storage-account-type
+ - location
+ - subscription-id
+ - application-password
+ - application-id
+ - tenant-id
+
+#### `storage-account-type`
+
+Use the generic value of 'Standard_LRS' for this.
+
+### `location`
+
+Choose an [Azure region](https://azure.microsoft.com/en-us/regions/). Here we will use
+'East US'.
+
+### `subscription-id`
+
+List your account and get the subscription ID, the **SUB_ID**:
 
-The generic configuration sections generated for Windows Azure will look
-something like this:
+```bash
+azure account list
+```
+
+Set its value in the terminal. It will look similar to:
+
+```bash
+SUB_ID=f717c8c1-8e5e-4d38-be7f-ed1e1c879e1a
+```
+
+### `application-password`
+
+You will create an application in the next step. For now, create a password for
+it, the **APP_PASSWORD**.
+
+```bash
+APP_PASSWORD=some_password
+```
+
+### `application-id`
+
+Create an Azure Active Directory (AAD) application:
+
+```bash
+azure ad app create \
+	--name "ubuntu.example.com" \
+	--home-page "http://ubuntu.example.com" \
+	--identifier-uris "http://ubuntu.example.com" \
+	--password $APP_PASSWORD
+```
+
+The options `--name`, `--home-page`, and `--identifier-uris` are arbitrary but
+you should use values that make sense for your environment.
+
+Note the application ID, the **APP_ID**. It will look similar to:
+
+```bash
+APP_ID=f6ab7cbd-5029-43ef-85e3-5c4442a00ba8
+```
+
+Use the APP_ID to create a server principal:
```

Comment:
what is a 'server principal'? I mean, we can guess, but is this handily defined somewhere on the Azure site?